client/src/components/products/GoogleMapsInput.js

```diff
@@ -0,0 +1,222 @@
+      if (addressArray[i].types[0] && "country" === addressArray[i].types[0]) {
+        country = addressArray[i].long_name;
+        return country;
+      }
+    }
+  };
+
+  onPlaceSelected = place => {
+    console.log(place);
+    if (place.formatted_address) {
+      const address = place.formatted_address,
+        addressArray = place.address_components,
+        city = this.getCity(addressArray),
+        country = this.getCountry(addressArray),
+        latValue = place.geometry.location.lat(),
+        lngValue = place.geometry.location.lng();
+
+      this.props.getMapData(address, city, country, latValue, lngValue);
+
+      // this.setState({
+      //   address: address ? address : "",
+      //   city: city ? city : "",
+      //   country: country ? country : "",
+      //   markerPosition: {
+      //     lat: latValue,
+      //     lng: lngValue
+      //   },
+      //   mapPosition: {
+      //     lat: latValue,
+      //     lng: lngValue
+      //   }
+      // });
+    } else {
+      Geocode.fromAddress(place.name)
+        .then(response => {
+          console.log(response);
+          const data = response.results[0];
+          const address = data.formatted_address,
+            addressArray = data.address_components,
+            city = this.getCity(addressArray),
+            country = this.getCountry(addressArray),
+            latValue = data.geometry.location.lat,
+            lngValue = data.geometry.location.lng;
+
+          this.props.getMapData(address, city, country, latValue, lngValue);
+
+          // this.setState({
+          //   address: address ? address : "",
+          //   city: city ? city : "",
+          //   country: country ? country : "",
+          //   markerPosition: {
+          //     lat: latValue,
+          //     lng: lngValue
+          //   },
+          //   mapPosition: {
+          //     lat: latValue,
+          //     lng: lngValue
+          //   }
+          // });
+        })
+        .then(err => {
+          console.log(err);
+        });
+    }
+  };
+
+  onMarkerDrag = event => {
+    console.log(event);
+    let newLat = event.latLng.lat(),
+      newLng = event.latLng.lng();
+
+    Geocode.fromLatLng(newLat, newLng)
+      .then(response => {
+        const address = response.results[0].formatted_address,
+          addressArray = response.results[0].address_components,
+          city = this.getCity(addressArray),
+          country = this.getCountry(addressArray);
+
+        this.props.getMapData(address, city, country, newLat, newLng);
+
+        // this.setState({
+        //   address: address ? address : "",
+        //   country: country ? country : "",
+        //   city: city ? city : "",
+        //   markerPosition: {
+        //     lat: newLat,
+        //     lng: newLng
+        //   },
+        //   mapPosition: {
+        //     lat: newLat,
+        //     lng: newLng
+        //   }
+        // });
```

Comment:
Do we still need this? If not, let's remove it?

Reply:
removed 🗑